crates/task-impls/src/quorum_proposal/dependency_handle.rs

```diff
@@ -185,6 +187,13 @@ impl<TYPES: NodeType> HandleDepOutput for ProposalDependencyHandle<TYPES> {
             .validated_state_map()
             .contains_key(&high_qc_view_number)
         {
+            // The proposal for the high qc view is missing, try to get it asynchronously
+            let memberhsip = Arc::clone(&self.quorum_membership);
+            let sender = self.sender.clone();
+            let consensus = Arc::clone(&self.consensus);
+            async_spawn(async move {
+                fetch_proposal(high_qc_view_number, sender, memberhsip, consensus).await
```

Comment:
How come this is an async spawn instead of just calling? It looks like the method doesn't do anything long-running?

Reply:
The fetch could be potentially very slow (500ms) and this caller isn't expecting the result directly back, in the mean time we might get the `ValidatedStateUpdated` event we need before the request for proposal ever even finishes